Sources/Core/AWSClientRuntime/EventStream/AWSMessageSigner.swift

```diff
@@ -10,22 +10,36 @@ import ClientRuntime
 extension AWSEventStream {
     /// Signs a `Message` using the AWS SigV4 signing algorithm
     public class AWSMessageSigner: MessageSigner {
-        let context: HttpContext
         let encoder: MessageEncoder
+        let signingConfig: () async throws -> AWSSigningConfig
+        let requestSignature: () -> String
 
-        private var previousSignature: String
+        private var _previousSignature: String?
 
-        public init(context: HttpContext, encoder: MessageEncoder) {
-            self.context = context
-            self.encoder = encoder
+        /// Returns the previous signature used to sign a message
+        /// If no previous signature is available, then the request signature returned
+        /// which acts as previous signature for the first message
+        var previousSignature: String {
```

Comment:
Is this just a lazy var? 
`lazy var previousSignature: String = requestSignature()`
This delays computation until the property is first accessed, and it also allows for the value to be explicitly set. 

Its a bit simpler than having the duplicate properties.

Reply:
could have been lazy if we need it only get, but we set also. hence need a backing field.